### PR TITLE
refactor(expo-updates): drop glob in favor of `npm pack --json`

### PR DESCRIPTION
**Pull request by byCedric**
# Why

- Part of the `glob` upgrade: https://github.com/expo/expo/pull/29808
- Missing package caused a failure in: https://github.com/expo/expo/actions/runs/9658577534/job/26640053473?pr=29898

~~In the future, we might be able to use Node's own `fs.globSync` for this. But I think it's still too early to start using it ([docs](https://nodejs.org/api/fs.html#fsglobsyncpattern-options))~~

Edit: [due to the path collision](https://github.com/expo/expo/pull/29985#issuecomment-2196843267), and the fact that we can't modify the filename for `npm pack`, I dropped glob in favor of parsing the output from `npm pack --json`. This also removes the need for glob.

# How

- Dropped `glob` in favor of `npm pack --json`

# Test Plan

See if CI passes

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by douglowder — APPROVED** [review]

**Comment by douglowder** [comment]
@byCedric no the CI failures are not expected... looks like the expo-updates CLI is breaking, and I'm seeing warnings about config-plugins using an old glob version.

**Comment by byCedric** [comment]
@douglowder Unfortunately, the `glob@7`/`glob@<9` warnings will still persist for a bit. We use it extensively, and I still need to update packages outside this repo 🙈 

But at least this change would get expo/expo back on a good glob version.

**Review by wschurman — CHANGES_REQUESTED** [review]
Not sure what's going on with the tests but it does seem to be due to this change. The instructions on how to test locally are at https://github.com/expo/expo/blob/main/packages/expo-updates/e2e/README.md. Happy to help if needed, but passing back to you for now.

**Comment by byCedric** [comment]
Oh, I see what's happening here! It's an unsafe glob return 😄

TL;DR;
- Before `glob@9`, glob results were sorted alphabetically.
- Since `glob@9`, glob results are not sorted, and returned as the system returns it

This is normally not a problem, but we are using `<dependency>-<version>-<timestamp>.tgz` as glob pattern. Unfortunately, the `-` separator between `<dependency>` and `<version>` may overlap with different results:

image

And, with the alphabetical sorting, we never hit this issue. Imho, we shouldn't rely on alphabetical sorting, but try to use a `<dependency>` and `<version>` separator that can never conflict with dependency names itself. E.g. something like `<dependency>@<version>` might be best here, since `@` is not allowed in package names on npm, but allowed as file name on Windows/MacOS/Linux.

**Review by wschurman — APPROVED** [review]

**Review by douglowder — APPROVED** [review]
Thanks!!!
